packages/text-input/src/index.tsx

```diff
@@ -1,33 +1,82 @@
 import * as React from 'react'
-import {TextInputAppearanceType, TextInputProps} from './types'
+import {TextInputAppearanceType, TextInputProps, AffixProps} from './types'
 import * as S from './styled'
 import {
   getTextSizeForControlHeight,
   getBorderRadiusForControlHeight,
   useDefaults
 } from '@smashing/theme'
 
+const Affix: React.FC<AffixProps> = ({
+  affix,
+  component: IconComponent,
+  disabled,
+  height,
+  inputRef,
+  invalid
+}) => (
+  <React.Fragment>
+    {affix === 'before' ? (
+      <S.InputBefore invalid={invalid} disabled={disabled} height={height}>
+        <IconComponent inputRef={inputRef} />
```

Comment:
We're not considering adding `inputRef` property to the generated icons nor their API, so clicking mechanism should be handled dynamically. Other idea is to get `ref` objects from icon svg's and handle their event listeners elsewhere